### PR TITLE
Publish ruler's ring metrics

### PR DESCRIPTION
**Pull request by lahsivjar**
Signed-off-by: lahsivjar <lahsivjar@gmail.com>

**What this PR does**: Currently ruler doesn't publish it's ring metrics since it is not registered. Also, the ruler ring is currently named as `ring` which has been changed in this MR to `ruler` so that the metrics is understandable.

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by lahsivjar** [comment]
@gouthamve Please take a look at the PR whenever you can

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 * [ENHANCEMENT] Query-tee: add support for doing a passthrough of requests to preferred backend for unregistered routes #3018
 * [ENHANCEMENT] Expose `storage.aws.dynamodb.backoff_config` configuration file field. #3026
 * [ENHANCEMENT] Added `cortex_request_message_bytes` and `cortex_response_message_bytes` histograms to track received and sent gRPC message and HTTP request/response sizes. Added `cortex_inflight_requests` gauge to track number of inflight gRPC and HTTP requests. #3064
+* [ENHANCEMENT] Publish ruler's ring metrics
```
Comment (pstibrany) [inline CHANGELOG.md:20]:
```suggestion
* [ENHANCEMENT] Publish ruler's ring metrics. #3074
```

File: pkg/ruler/ruler.go [hunk 2]
```diff
@@ -235,7 +236,7 @@ func enableSharding(r *Ruler, ringStore kv.Client) error {
 		return errors.Wrap(err, "failed to initialize ruler's lifecycler")
 	}
 
-	r.ring, err = ring.NewWithStoreClientAndStrategy(r.cfg.Ring.ToRingConfig(), ring.RulerRingKey, ring.RulerRingKey, ringStore, &ring.DefaultReplicationStrategy{})
+	r.ring, err = ring.NewWithStoreClientAndStrategy(r.cfg.Ring.ToRingConfig(), "ruler", ring.RulerRingKey, ringStore, &ring.DefaultReplicationStrategy{})
```
Comment (pstibrany) [inline pkg/ruler/ruler.go:239]: Please also change ring name for lifecycler few lines above.

File: pkg/ruler/ruler.go [hunk 3]
```diff
@@ -212,6 +212,7 @@ func NewRuler(cfg Config, manager MultiTenantManager, reg prometheus.Registerer,
 		if err = enableSharding(ruler, ringStore); err != nil {
 			return nil, errors.Wrap(err, "setup ruler sharding ring")
 		}
+		prometheus.MustRegister(ruler.ring)
```
Comment (pstibrany) [inline pkg/ruler/ruler.go:215]:
```suggestion
        if reg != nil {
            reg.MustRegister(ruler.ring)
        }
```

**Review by pstibrany — COMMENTED** [review]
Thank you for your contribution! It looks good, with tiny minor comments.

Would you please rebase this PR on top of master, as there were some conflicting changes in gossip integration test? Thanks!

**Comment by lahsivjar** [comment]
> Would you please rebase this PR on top of master, as there were some conflicting changes in gossip integration test? 

@pstibrany Fixed all the comments and rebased.

**Review by pstibrany — APPROVED** [review]
Thank you!

**Review by bboreham — APPROVED** [review]
Thanks!

**Review by pracucci — APPROVED** [review]
Yes yes yes, thanks!

**Pull request merged**